### ISSUE TITLE
Add a `--pull` option to`balena build`.

### ISSUE THREAD

**Issue by jimsynz** [balena-io/balena-cli#1584]
- **balena CLI version:** 11.21.7
- **Operating system version:** macOS Catalina 10.15.2
- **32/64 bit OS and processor:** 64 bit 6-core i7
- **Install method:** executable installer

---

**TLDR:** support `balena build --app myApp --pull` to ensure that the base images are up to date.

---

While working on support today I noticed that it's non-trivial to ensure that you're using the latest versions of the base images that your project depends on when using the `balena build` workflow.  Additionally, I thought that the `--nocache` option may provide this functionality, but it does not (this is good, because I think that this would be confusing).

Manually pulling the base images is a little more difficult because of the templating which we support for device type images.  The only way I can think of to ensure that you have the latest versions of the base images involves running the build twice:

```sh
balena build --application myApp --logs | awk '/FROM/{print $7}' | xargs -n 1 docker pull
balena build --application myApp
```

**Comment by pdcastro** [comment]
It makes sense, especially as I see that `docker build` has such an option:
> https://docs.docker.com/engine/reference/commandline/build/
> `--pull    Always attempt to pull a newer version of the image`

**Comment by kfrance** [comment]
This is also an issue with push to a local device. I've updated the base image but I can't get the push to update the base image and --nocache doesn't help.

**Comment by pdcastro** [comment]
This issue was implemented in CLI release v12.16.0.
